dartagnan/src/main/java/com/dat3m/dartagnan/program/processing/Inlining.java

```diff
@@ -39,6 +41,9 @@ public class Inlining implements FunctionProcessor {
             secure = true)
     private int bound = 1;
 
+    // This map is updated when a new program is passed to this instance.
+    private final Map<Function, CacheEntry> cache = new HashMap<>();
+
```

Comment:
Is the cache necessary for correctness (maybe because you need to cache the original state of a function)?
If so, then maybe this pass should be changed back to a `ProgramProcessor` so it can properly traverse all functions before transformation.
Because as it stands, the cache has two flaws:
- Rerunning the pass on the same program can easily break if additional function calls get inserted afterwards (by some other pass).
- The cache can take up memory indefinitely (depending on whether references to `Inlining` survive). A potential fix would be to use a weak hash map (or some Guava cache with eviction policies), though it has the potential to evict entries during processing which would break the pass. Using a Guava cache with time-based eviction policy would be workable in practice though.

Reply:
What was bugging me while testing was that `Inlining` was not parallelizable along the functions.  Maybe it should indeed be a `ProgramProcessor`.  For indirect recursion, the sequencing of `Inlining` means that the minimal recursion bound for finding a bug differs by 1 based on the definition order in the program.  Also, I could not find a better naming scheme for registers and labels.  `Inlining` names registers like `1:1:6:3:2:r8` to avoid name conflicts.
The version with caches helped me with debugging, but now I managed to fix the remaining bugs.  Along with the cache, I made use of an event tag to exclude writers to the result register of the call when renaming registers.  The new commit got rid of both again.

Reply:
The cache's lifetime is now completely inside `run(Program)`.  With this, `Inlining` should be reusable again.